docs/cloud/satellites/self-hosted.md

```diff
@@ -0,0 +1,215 @@
+my-satellite  linux/amd64  self-hosted  v0.8.3   operational
+...
+```
+
+The size field of a self-hosted satellite will be listed as `self-hosted`, as opposed to one of the sizes of Earthly Cloud’s managed satellites.
+
+The state field will be either `operational` or `offline`. A self-hosted satellite automatically transitions to an `offline` state when it is gracefully terminated, or if no heartbeat has been received by Earthly’s servers for a while.
+
+You can run `satellite rm` on a self-hosted satellite when it is in an offline state to remove it from your account permenantly.
+
+## Platform-Specific Guides
+
+### Kubernetes
+
+Below is a basic example of how to start a self-hosted satellite as a Kubernetes Pod:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-satellite
+  namespace: earthly-satellites
+spec:
+  containers:
+   - name:
+   valueFrom:
+     fieldRef:
+       fieldPath: metadata.name
+      image: earthly/satellite:v0.8.3
+      securityContext:
+        privileged: true
+      ports:
+        - containerPort: 8372
+       env:
+        - name: EARTHLY_ORG
+          value: my-org
+        - name: EARTHLY_TOKEN
+          value: u4a*************l92
+        - name: SATELLITE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: SATELLITE_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.podIP
+```
+
+Note that this example uses the Pod’s IP address (via [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/)) as the `SATELLITE_HOST` value. The Earthly CLI must be able to reach the IP on its network.
+
+If the Earthly CLI should connect via a different address, such as DNS, then this value should be provided as the `SATELLITE_HOST` instead.
+
+
+### AWS EC2
+When launching on EC2, we recommend using the latest version of Amazon Linux 2023. The following cloud-init script can be configured when launching a new EC2 instance so that it automatically starts the satellite on boot.
+
+```yaml
+#cloud-config
+runcmd:
+  - sudo dnf update
+  - sudo dnf install -y docker
+  - sudo systemctl start docker.service
+  - sudo systemctl enable docker.service
+  - |-
+      sudo docker run -d --privileged \
+        --restart always \
+        --name satellite \
+        -p 8372:8372 \
+        -v /earthly-cache:/tmp/earthly:rw \
+        -e EARTHLY_TOKEN=GuFna*****nve7e \
+        -e EARTHLY_ORG=my-org \
+        -e SATELLITE_NAME=my-satellite \
+        earthly/satellite:v0.8.3
+```
+
+Note that the `SATELLITE_HOST` variable is unset in this example so that the host is auto-discovered by the satellite when it starts. This should result in the instance’s private DNS being used as the host. 
+
+If the Earthly CLI will be unable to connect to the satellite via the EC2’s private DNS, then `SATELLITE_HOST` should be provided in the `docker run` command with an alternate value.
+
+## Advanced Configuration
+
+### Cache Disk
```

Comment:
I think that our position should be stronger here. Not using a volume causes extremely bad `WITH DOCKER` performance (due to overlayfs). We should warn users that they MUST use a volume for this, and that volume should be backed by an ext4 filesystem (Docker volumes work great).

All our examples above should include the volume mount (the minimal docker run command and k8s example).

Reply:
I didn't realize there were legitimate issues with not using a volume. I'll put this in the other examples as well then.